### PR TITLE
[PM-1900] Run Process Reload on Firefox

### PR DESCRIPTION
**Pull request by justindbaur**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-1900
## 📔 Objective

Switch Firefox from running `window.location.reload(true)` to `chrome.runtime.reload()`. Also to help avoid the sidebar from automatically re-opening when the process reload happens we are not going to open it automatically on install (since process reload emulates a reinstall). The sidebar is still open-able through the shortcut.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by djsmith85 — DISMISSED** [review]
Changes are looking good, as they are quite similar to https://github.com/bitwarden/clients/pull/3683 😉 

Could please check if
- apps/browser/src/popup/app.component.ts: Removed a message handler for "reloadProcess" which never got called.
from https://github.com/bitwarden/clients/pull/3683 still applies and also remove it here.

**Comment by justindbaur** [comment]
@djsmith85 Oh my word, I totally dropped the ball on the PR? Would you rather revive that?

I did some looking and the `"reloadProcess"` message is sent from SystemService but I don't think it would be necessary anymore. So I am happy to remove it as well. I did test the `open_at_install` & `chrome.runtime.reload` behavior in Opera and couldn't notice any adverse effects. I didn't consider safari though, I figured it didn't call it because the API wasn't supported, since as far as I knew they didn't have a sidebar. I will do some testing on it though and make those changes here.

**Comment by djsmith85** [comment]
> @djsmith85 Oh my word, I totally dropped the ball on the PR? Would you rather revive that?
> 
> I did some looking and the `"reloadProcess"` message is sent from SystemService but I don't think it would be necessary anymore. So I am happy to remove it as well. I did test the `open_at_install` & `chrome.runtime.reload` behavior in Opera and couldn't notice any adverse effects. I didn't consider safari though, I figured it didn't call it because the API wasn't supported, since as far as I knew they didn't have a sidebar. I will do some testing on it though and make those changes here.

No worries, at lot has happened since. Several changes to the process reload (clean up, refactor, biometrics) and more importantly MV3. Hence all the merge conflicts. I'm happy with keeping this open and close mine. Looking forward to hear about your findings when testing this and let me know if you wanna chat about anything.

File: apps/browser/src/manifest.v3.json [hunk 1]
```diff
@@ -137,7 +137,8 @@
   "sidebar_action": {
     "default_title": "Bitwarden",
     "default_panel": "popup/index.html?uilocation=sidebar",
-    "default_icon": "images/icon19.png"
+    "default_icon": "images/icon19.png",
+    "open_at_install": false
```
Comment (coroiu) [inline apps/browser/src/manifest.v3.json:141]: omg thank you this annoys me so much!

**Review by coroiu — APPROVED** [review]

**Review by coroiu — COMMENTED** [review]
failing tests

File: apps/browser/src/platform/browser/browser-api.ts [hunk 2]
```diff
@@ -414,16 +414,8 @@ export class BrowserApi {
   /**
    * Handles reloading the extension, either by calling the window location
    * to reload or by calling the extension's runtime to reload.
```
Comment (djsmith85) [inline apps/browser/src/platform/browser/browser-api.ts:416]: ⚠️ Documentation needs to be updated

**Review by djsmith85 — CHANGES_REQUESTED** [review]
Changes are looking great, just one minor documentation change

File: apps/browser/src/platform/browser/browser-api.ts [hunk 3]
```diff
@@ -412,18 +412,9 @@ export class BrowserApi {
   }
 
   /**
-   * Handles reloading the extension, either by calling the window location
-   * to reload or by calling the extension's runtime to reload.
-   *
-   * @param globalContext - The global context to use for the reload.
+   * Handles reloading the extension using underlying functionalify exposed by the browser API.
```
Comment (djsmith85) [inline apps/browser/src/platform/browser/browser-api.ts:415]:
```suggestion
   * Handles reloading the extension using the underlying functionality exposed by the browser API.
```

**Review by djsmith85 — DISMISSED** [review]

**Review by coroiu — DISMISSED** [review]

**Review by djsmith85 — APPROVED** [review]

**Review by coroiu — DISMISSED** [review]

**Review by djsmith85 — APPROVED** [review]

**Pull request merged**